### PR TITLE
[mce-plugin] Add lock state tracking to the plugin. JB#56983

### PR DESCRIPTION
**Pull request by ikamaletdinov**
This change also disallow to use NFC without unlocking the device.
(like in the Android)

This behaviour could be enabled only in build time adding
`nfc_disable_in_locked 1` macro. Disabled by default.

### CODE REVIEW

**Comment by spiiroin** [comment]
> This change also disallow to use NFC without unlocking the device.
> (like in the Android)

Looks like it is tracking "touchscreen and keypad lock" status, rather than "device lock" status and I'm wondering if that is intentional / matches that "like in Android" bit. For example: it is possible/normal to have tklock=false while display is off - unless Settings -> Device lock -> Automatic locking = No delay (in which case "device lock" is applied whenever display is turned off, and that activates "tklock" too).

**Comment by monich** [comment]
> Looks like it is tracking "touchscreen and keypad lock" status, rather than "device lock" status and I'm wondering if that is intentional / matches that "like in Android" bit. For example: it is possible/normal to have tklock=false while display is off - unless Settings -> Device lock -> Automatic locking = No delay (in which case "device lock" is applied whenever display is turned off, and that activates "tklock" too).

Hmm... What would be an example of "device lock" without "touchscreen and keypad lock"? Or "touchscreen and keypad lock" without "device lock". To me, those are interchangeable terms. The device is locked when you can't access the home screen without entering the lock code (or passing the fingerprint check) and vice versa. Or am I missing something...

File: README [hunk 1]
```diff
@@ -1,2 +1,5 @@
 nfcd plugin for mce-based device state tracking.
 Keeps NFC off when display is off.
+
+It is also possible to disable NFC in locked state also:
+Just add `nfc_disable_in_locked 1` macro during build.
```
Comment (monich) [inline README:5]: I think this is an unnecessary detail, especially considering that this is not the final solution. Eventually, this option should (and hopefully will) be configurable at run time.

**Comment by spiiroin** [comment]
> Hmm... What would be an example of "device lock" without "touchscreen and keypad lock"?

There is a before/after sfos 2 divide when unlock ui (="Enter security code" -view) became part of locksreen ui. So, in sfos 1 unlocking device had to happen in devicelock=locked and tklock=unlocked state whereas from sfos 2 onwards such state is not allowed (devicelock transition to locked state -> tklock=locked too; setting tklock=unlocked is not allowed while devicelock=locked).

> Or "touchscreen and keypad lock" without "device lock". To me, those are interchangeable terms. The device is locked when you can't access the home screen without entering the lock code (or passing the fingerprint check) and vice versa. Or am I missing something...

The most obvious case is: when device lock code is not used (Settings -> Device lock -> Automatic locking = Not in use) -> devicelock is always in unlocked state, but lockscreen ui still exists and is used -> tklock has a life of its own.

Basically: when you see lockscreen / unlock ui -> tklock is locked. And if you can dismiss lockscreen without providing lock code / fingerprint -> devicelock must already be in unlocked state.

**Comment by ikamaletdinov** [comment]
Replaced with: https://github.com/mer-hybris/nfcd-mce-plugin/pull/3
